cmd/podmanV2/containers/top.go

```diff
@@ -0,0 +1,91 @@
+package containers
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/containers/libpod/cmd/podmanV2/registry"
+	"github.com/containers/libpod/pkg/domain/entities"
+	"github.com/containers/psgo"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	topDescription = fmt.Sprintf(`Similar to system "top" command.
+
+  Specify format descriptors to alter the output.
+
+  Running "podman top -l pid pcpu seccomp" will print the process ID, the CPU percentage and the seccomp mode of each process of the latest container.
+  Format Descriptors:
+    %s`, strings.Join(psgo.ListDescriptors(), ","))
+
+	topOptions = entities.TopOptions{}
+
+	topCommand = &cobra.Command{
+		Use:   "top [flags] CONTAINER [FORMAT-DESCRIPTORS|ARGS]",
+		Short: "Display the running processes of a container",
+		Long:  topDescription,
+		RunE:  top,
+		Args:  cobra.ArbitraryArgs,
+		Example: `podman top ctrID
+podman top --latest
+podman top ctrID pid seccomp args %C
+podman top ctrID -eo user,pid,comm`,
+	}
+)
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Mode:    []entities.EngineMode{entities.ABIMode, entities.TunnelMode},
+		Command: topCommand,
+	})
+
+	flags := topCommand.Flags()
+	flags.SetInterspersed(false)
+	flags.BoolVar(&topOptions.ListDescriptors, "list-descriptors", false, "")
+	flags.BoolVarP(&topOptions.Latest, "latest", "l", false, "Act on the latest container podman is aware of")
+
+	_ = flags.MarkHidden("list-descriptors") // meant only for bash completion
+}
+
+func top(cmd *cobra.Command, args []string) error {
+	if topOptions.ListDescriptors {
+		fmt.Println(strings.Join(psgo.ListDescriptors(), "\n"))
+		return nil
+	}
+
+	if len(args) < 1 && !topOptions.Latest {
+		return errors.Errorf("you must provide the name or id of a running container")
+	}
+
+	// FIXME: Should we really call NewContainerEngine here?
+	//        Should main take care of that?
+	_, err := registry.NewContainerEngine(cmd, args)
```

Comment:
dont need this ... add in the cobra command as 
```
PersistentPreRunE: preRunE,
```

Reply:
Perfect, worked. Thanks!